component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAuthenticator.java

```diff
@@ -0,0 +1,881 @@
+        username when the user is provisioned to the server.
+         */
+        String mappedLocalUsername = getMappedLocalUsername(authenticatedUserFromContext, context);
+
+        /*
+        If the mappedLocalUsername is blank, that means this is an initial login attempt by a non-provisioned
+        federated user.
+         */
+        boolean isInitialFederationAttempt = StringUtils.isBlank(mappedLocalUsername);
+
+        try {
+            AuthenticatedUser authenticatingUser = resolveAuthenticatingUser(context, authenticatedUserFromContext,
+                    mappedLocalUsername, tenantDomain, isInitialFederationAttempt);
+            username = UserCoreUtil.addTenantDomainToEntry(authenticatingUser.getUserName(), tenantDomain);
+            context.setProperty(BackupCodeAuthenticatorConstants.AUTHENTICATED_USER, authenticatingUser);
+
+            String retryParam = "";
+            if (context.isRetrying()) {
+                retryParam = "&authFailure=true&authFailureMsg=login.fail.message";
+            }
+            boolean isBackupCodesExistForUser = false;
+            boolean isBackupCodesEnabledByUser = false;
+            // Not required to check the backup code enable state for the initial login of the federated users.
+
+            if (!isInitialFederationAttempt) {
+                isBackupCodesExistForUser = isBackupCodesExistForUser(UserCoreUtil.addDomainToName(username,
+                        authenticatingUser.getUserStoreDomain()));
+                isBackupCodesEnabledByUser = isBackupCodesEnabledByUser(UserCoreUtil.addDomainToName(username,
+                        authenticatingUser.getUserStoreDomain()));
+            }
+            if (isBackupCodesExistForUser) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Backup codes exists for the user: " + username);
+                }
+            }
+            boolean isBackupCodesEnabledByAdmin = IdentityHelperUtil.checkSecondStepEnableByAdmin(context);
+            if (log.isDebugEnabled()) {
+                log.debug("Backup Code  is enabled by admin: " + isBackupCodesEnabledByAdmin);
+            }
+            // This multi option URI is used to navigate back to multi option page to select a different
+            // authentication option from backup code pages.
+            String multiOptionURI = BackupCodeUtil.getMultiOptionURIQueryParam(request);
+
+            if (isBackupCodesExistForUser &&
+                    request.getParameter(BackupCodeAuthenticatorConstants.ENABLE_BACKUP_CODE) == null) {
+                //if backup code is enabled for the user.
+                String backupCodeLoginPageUrl =
+                        buildBackupCodeLoginPageURL(context, username, retryParam, multiOptionURI);
+                response.sendRedirect(backupCodeLoginPageUrl);
+            } else {
+
+                if (isBackupCodesEnabledByAdmin) {
```

Comment:
Remove this condition as we have no such thing.

Reply:
Fixed